Django1.8

vrocha opened the pull request:
Makes the changes necessary so that the project can be run using django 1.8.7

One of the biggest changes is that south migrations are no longer needed and instead we need to create native django migrations. There are a few extra steps that need to be taken to covert one existing project to 1.8 when it comes to migrations. 

First, django built-in apps need to be migrated: python manage.py migrate --fake
Then, python manage.py migrate --fake-initial

One only needs to run these commands once, there after the usual manage.py migrate will do the work.

vrocha committed:
[django1.8] Initial work

vrocha committed:
[django1.8] Explorer migrations changes

vrocha committed:
[django1.8] Create new migration for django

vrocha committed:
[django 1.8] Remove empty file

vrocha committed:
[django1.8] Fixes issues with "admin" site

vrocha committed:
[django1.8] update documentation

vrocha committed:
[django1.8] Remove south from requirements

hlapp commented:
@victor-rocha do you intend to hold this and merge on your own? The diff is way too big for me to do any review on, so by default I'm assuming that you are holding this for yourself here for now. That said, I'm happy to go ahead and merge as well.

victor-rocha commented:
@hlapp Yes, this is still on hold. I want to make sure that regression testing goes as expected and then I will be more confident about merging.

vrocha committed:
[django 1.8] Fixes select_related on non relational field

vrocha committed:
[django1.8] Adds public directory for static assets

vrocha committed:
[django1.8] Settings to server static content correctly

vrocha committed:
[django1.8] Move static directory outside project root.

This is not really necessary but a good security measure
given our server configuration.

vrocha committed:
[django1.8] Fixes collectstatic documentation

vrocha committed:
[django1.8] One more edit to how to collect static

vrocha committed:
[django1.8] Include apache alias for media/static assets

vrocha committed:
[django1.8] Fixes form id issue.

Styles wouldn't apply after upgrade because of deprecated
`module_name` opts attribute.

vrocha committed:
[django1.8] Use django's StreamingHTTPResponse instead of HTTPResponse
